src/org/jitsi/impl/neomedia/MediaStreamImpl.java

```diff
@@ -611,7 +611,7 @@ private boolean addReceiveStream(ReceiveStream receiveStream)
             {
                 MediaDeviceSession deviceSession = getDeviceSession();
 
-                if (deviceSession == null || rtpTranslator != null)
+                if (deviceSession == null)
                 {
```

Comment:
Are you sure this fixes your problem? The rtpTranslator is used only in jvb and when configured in jigasi, otherwise it is always null for Jitsi Desktop and should not matter.
This was added because of jigasi and this change my break it, I think ...

Reply:
Yes, definitely the cause, double and triple checked, just that condition. With that additional check, the Java Media Framework (JMF) is not creating any RTPStream for incoming video.

Reply:
Hum, maybe you are right and this is not needed. As jigasi is in the same state with deviceSession not null...

Reply:
Actually I'm puzzled that this change breaks anything.  The whole change is about translator and in jitsi there is no translator, so even the check above is always checking just deviceSession variable.

Reply:
I have no idea about that part of the source code. If you need anything, like the parameter states, to de-puzzle that, please, do not hesitate to ask. I placed a `new Throwable().printStackTrace();` at that line and got this:
```
INFO: [117] net.sf.fmj.media.Log.info() Resetting queue, last seq added: 9223372036854775806, current seq: 63488
org.jitsi.impl.neomedia.MediaStreamImpl.addReceiveStream(MediaStreamImpl.java:613)
org.jitsi.impl.neomedia.MediaStreamImpl.update(MediaStreamImpl.java:3453)
org.jitsi.impl.neomedia.rtp.translator.RTPTranslatorImpl.update(RTPTranslatorImpl.java:1099)
net.sf.fmj.media.rtp.RTPEventHandler.processEvent(RTPEventHandler.java:92)
net.sf.fmj.media.rtp.RTPEventHandler.dispatchEvents(RTPEventHandler.java:50)
net.sf.fmj.media.rtp.RTPEventHandler.run(RTPEventHandler.java:114)
INFO: [160] net.sf.fmj.media.Log.info() Starting RTPSourceStream.
INFO: [163] org.jitsi.impl.neomedia.codec.video.vp8.DePacketizer.log() Opened VP8 depacketizer
```
I guess that nobody noticed that, yet, because that more obvious HFlip issue let anyone think that video does not work in general.